src/App.tsx

```diff
@@ -1,12 +1,75 @@
+/* eslint-disable no-console */
 import React from 'react';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { GoodsList } from './components/GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+type State = {
+  goods: Good[];
+  isLoading: boolean;
+  loadingErr: boolean;
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: [],
+    isLoading: false,
+    loadingErr: false,
+  };
+
+  loadGoods = async (request: () => Promise<Good[]>) => {
+    this.setState({
+      goods: [],
+      loadingErr: false,
+      isLoading: true,
+    });
+
+    try {
+      const goods = await request();
+
+      this.setState({
+        goods,
+      });
+    } catch (error) {
+      this.setState({ loadingErr: true });
+    } finally {
+      this.setState({ isLoading: false });
+    }
+  };
+
+  render(): React.ReactNode {
+    const { goods, loadingErr, isLoading } = this.state;
+
+    return (
+      <div className="App">
+        <h1 className="App__title">Dynamic list of Goods</h1>
+        <div className="App__buttons">
+          <button type="button" onClick={() => this.loadGoods(getAll)}>
+            Load All goods
+          </button>
+          <button type="button" onClick={() => this.loadGoods(get5First)}>
+            Load 5 first goods
+          </button>
+          <button type="button" onClick={() => this.loadGoods(getRedGoods)}>
+            Load red goods
+          </button>
+          {console.log(this.state.goods)}
+        </div>
+        <div className="App__content">
+          {isLoading && (
+            <div className="App__content-loading">Loading... Please, wait</div>
+          )}
+          {goods.length !== 0 && (
```

Comment:
```suggestion
          {goods.length && (
```

Reply:
![image](https://user-images.githubusercontent.com/91382085/146821370-8491c456-1225-4274-8387-8905000d3238.png)
In this case i see 0 while loading the list, so I replace that with the following: ``!!goods.length``